### PR TITLE
Update code consistency

### PR DESCRIPTION
**Pull request by threeang**
#### Summary
Update the way to access properties in the section "Active learning: Working through a JSON example".

Before changes: Both dot notation and bracket notation appeared in the code.
After changes: Only dot notation will appear in the code.

#### Motivation
Improve code consistency and readability, as ESLint enforces dot notation style whenever possible.

### CODE REVIEW

**Comment by threeang** [comment]
Hi @teoli2003, I noticed that you removed yourself as a reviewer, I'm not sure if this Pull Request is good or not.

I understand that [this article](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/JSON) is for beginners, maybe it wants to inform beginners there are two ways to access object properties, although I think it should just use dot notation like this PR to keep the code consistent.

Could you give me some suggestions to improve this PR like should [the article](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/JSON) stay original (close this PR) or merge this?

Thank you.

**Comment by teoli2003** [comment]
I removed myself as a reviewer as I'm not familiar enough with the _Learning area_ to judge this.

**Comment by threeang** [comment]
> I removed myself as a reviewer as I'm not familiar enough with the _Learning area_ to judge this.

Okay, thanks.

**Comment by threeang** [comment]
> Just so that you know, it is the holiday period (read Summer in the Northern Hemisphere) and a lot of reviewers take time off in July or August, so review time may be slightly higher than usual.

I see, thanks a lot.

**Comment by hamishwillee** [comment]
Thanks @threeang. 

I'm going to merge this on the basis that it is certainly no worse.

ESLINT is opinionated, it [prefers dot notation because](https://eslint.org/docs/latest/rules/dot-notation#:~:text=In%20JavaScript%2C%20one%20can%20access,better%20with%20aggressive%20JavaScript%20minimizers.):

> ... the dot notation is often preferred because it is easier to read, less verbose, and works better with aggressive JavaScript minimizers.

That's "often", not always.  IFF this is important, then I think that we should consider updating https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Basics to note which is preferred and when. More specifically to note the cases when you prefer bracket notation (i.e. when the object is determined at runtime so you need to pass in the name as a string).

@teoli2003 Not strictly modernisation, but we should perhaps consider using dot notation as a default. Eventually we want to run ESLINT on examples, so this is a good idea.

**Review by hamishwillee — APPROVED** [review]

**Comment by teoli2003** [comment]
> @teoli2003 Not strictly modernisation, but we should perhaps consider using dot notation as a default. Eventually we want to run ESLINT on examples, so this is a good idea.

Yes, I think so too. Strangely, this preference is not documented anywhere. You are right: Eslint will help us find the few occurrences that are not following this. As I think this is uncontroversial and I'm adding it to the list (under already a rule but not followed 100% of the time.

**Comment by threeang** [comment]
@hamishwillee Thanks for the review and added link to [the ESLint dot-notation rule](https://eslint.org/docs/latest/rules/dot-notation).

File: files/en-us/learn/javascript/objects/json/index.md [hunk 1]
```diff
@@ -120,15 +117,19 @@ superHeroes['members'][1]['powers'][2]
 
 1. First we have the variable name — `superHeroes`.
 2. Inside that we want to access the `members` property, so we use `["members"]`.
-3. `members` contains an array populated by objects. We want to access the second object inside the array, so we use `[1]`.
-4. Inside this object, we want to access the `powers` property, so we use `["powers"]`.
-5. Inside the `powers` property is an array containing the selected hero's superpowers. We want the third one, so we use `[2]`.
+3. `members` contains an array populated by objects.
+   We want to access the second object inside the array, so we use `[1]`.
+5. Inside this object, we want to access the `powers` property, so we use `["powers"]`.
```
Comment (OnkarRuikar) [inline files/en-us/learn/javascript/objects/json/index.md:122]: Number 4 has been skipped.

Comment (threeang) [reply files/en-us/learn/javascript/objects/json/index.md:122]: resolved in https://github.com/mdn/content/commit/2039591f80b7218cbc4f5918f2a873814e83321c#diff-0762b941307d9e1d3f16894f7a2bc401d8f3a90afdefd12b03988f14acadc95c

Comment (hamishwillee) [reply files/en-us/learn/javascript/objects/json/index.md:122]: FYI, markdown supports autonumbering using `1.` for all items. @OnkarRuikar we might consider mandating that. It makes this kind of error a lot harder to make. 

There is a cost that you have to be careful when referring to numbered items - but that is always dangerous.

Comment (OnkarRuikar) [reply files/en-us/learn/javascript/objects/json/index.md:122]: Skipped numbers do get auto corrected during parsing to html. But while writing markdown, author may refer to a wrong position in description that follows.

This one failed the lint rule [MD029](https://github.com/DavidAnson/markdownlint/blob/v0.26.0/doc/Rules.md#md029). We can test this on [markdownlint playground](https://dlaa.me/markdownlint/).

**Pull request merged**